pkg/controllers/work/apply_controller.go

```diff
@@ -663,9 +663,22 @@ func (r *ApplyWorkReconciler) Leave(ctx context.Context) error {
 		klog.ErrorS(err, "Failed to list all the work object", "clusterNS", r.workNameSpace)
 		return client.IgnoreNotFound(err)
 	}
-	// we leave the resources on the member cluster for now
+	
+	// When leaving, we want to ensure resources placed on the member cluster remain there
```

Comment:
Please check how we handle the binding behavior when member is leaving.

Reply:
If i understand it correctly, we delete MemberCluster, Scheduler will detect this change and remove the binding, which will lead to remove the workloads placed by the work.

On the other side, MC deletion -> mark IMC as leave, memberAgent then will call this Leave func.

We cannot guarantee the order in this case.